src/Grid.cpp

```diff
@@ -0,0 +1,322 @@
+//
+//  Grid.cpp
+//  Game of Life
+//
+//  Created by Alex Frasca on 01/05/2023.
+//
+
+#include "Grid.hpp"
+
+namespace conway {
+
+// Cell
+
+Cell::Cell(sf::Vector2<int> idx, sf::Vector2<float> pos, float spc, CELL_STATE st) {
+    index = idx;
+    position = pos;
+    spacing = spc;
+    state = st;
+    kill_next_cycle = false;
+    activate_next_cycle = false;
+    
+    //init cell drawable
+    drawable = sf::ConvexShape();
+    drawable.setPointCount(4);
+    drawable.setPoint(0, sf::Vector2f( position.x          , position.y           ));
+    drawable.setPoint(1, sf::Vector2f( position.x + spacing, position.y           ));
+    drawable.setPoint(2, sf::Vector2f( position.x + spacing, position.y + spacing ));
+    drawable.setPoint(3, sf::Vector2f( position.x          , position.y + spacing ));
+}
+
+void Cell::tick() {
+    // change colors based on state for style
+    switch (state) {
+        case CELL_STATE::CELL_OFF:
+            drawable.setOutlineThickness(0);
+            drawable.setFillColor(GoL_Black);
+            break;
+        case CELL_STATE::CELL_LOW:
+            drawable.setOutlineColor(GoL_NavyBlue);
+            drawable.setOutlineThickness(-2);
+            drawable.setFillColor(GoL_DarkFucshia);
+            break;
+        case CELL_STATE::CELL_MEDIUM:
+            drawable.setOutlineColor(GoL_DarkFucshia);
+            drawable.setOutlineThickness(-2);
+            drawable.setFillColor(GoL_Fucshia);
+            break;
+        case CELL_STATE::CELL_HIGH:
+            drawable.setOutlineColor(GoL_Fucshia);
+            drawable.setOutlineThickness(2);
+            drawable.setFillColor(GoL_Goldenrod);
+            break;
+        case CELL_STATE::CELL_ON:
+            drawable.setOutlineColor(GoL_Goldenrod);
+            drawable.setOutlineThickness(2);
+            drawable.setFillColor(GoL_White);
+            break;
+        default:
+            break;
+    }
+    set_state();
+}
+
+void Cell::set_state() {
+    if(kill_next_cycle && !activate_next_cycle) {
+        decrement();
+        kill_next_cycle = false;
+    }
+    if(activate_next_cycle) {
+        state = CELL_STATE::CELL_ON;
+        activate_next_cycle = false;
+    }
+}
+
+void Cell::decrement() {
+    if(state != CELL_STATE::CELL_OFF) {
+        state = (CELL_STATE)((int)state - 1);
+    }
+}
+
+void Cell::update_position() {
+    position.x = index.x * spacing;
+    position.y = index.y * spacing;
+    drawable.setPoint(0, sf::Vector2f( position.x          , position.y           ));
+    drawable.setPoint(1, sf::Vector2f( position.x + spacing, position.y           ));
+    drawable.setPoint(2, sf::Vector2f( position.x + spacing, position.y + spacing ));
+    drawable.setPoint(3, sf::Vector2f( position.x          , position.y + spacing ));
+}
+
+// Grid
+
+Grid::Grid() {
+    
+    grid_width = 0;
+    grid_height = 0;
+    spacing = DEFAULT_SPACING;
+    
+}
```

Comment:
Not needed, initialize these values in the class definition and mark `Grid() = default` inside the class itself.